src/common/pony/detail/atomics.h

```diff
@@ -119,17 +117,18 @@ namespace ponyint_atomics
 namespace ponyint_atomics
 {
   template <typename T>
-  inline PONY_ABA_PROTECTED(T) big_load(PONY_ABA_PROTECTED(T)* ptr)
+  inline PONY_ABA_PROTECTED_PTR(T) big_load(PONY_ABA_PROTECTED_PTR(T)* ptr)
   {
-    PONY_ABA_PROTECTED(T) ret = {NULL, 0};
+    PONY_ABA_PROTECTED_PTR(T) ret = {NULL, 0};
```

Comment:
That would work, yes. I think it will only be possible in `pool_push/pull` and not in `pool_block_pull/push` because in the latter ones the CAS can be on any element of the list and if we fail, we always retry from the start of the list. I'll update the PR.